app/org/sagebionetworks/bridge/dynamodb/DynamoSurveyDao.java

```diff
@@ -351,16 +360,19 @@ public Survey getSurveyMostRecentlyPublishedVersion(StudyIdentifier studyIdentif
         return new QueryBuilder().setStudy(studyIdentifier).isPublished().setSurvey(guid).isNotDeleted().getOne(true);
     }
     
+    // SCAN
     @Override
     public Survey getSurveyMostRecentlyPublishedVersionByIdentifier(StudyIdentifier studyIdentifier, String identifier) {
         return new QueryBuilder().setStudy(studyIdentifier).setIdentifier(identifier).isPublished().isNotDeleted().getOne(true);
     }
```

Comment:
Long-term TODO, but this could probably be accomplished with an index. Many of the other "get one" calls could be converted to queries as well.